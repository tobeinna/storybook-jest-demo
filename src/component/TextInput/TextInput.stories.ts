import type { Meta, StoryObj } from "@storybook/react";
import { fn } from "@storybook/test";
import TextInput from "./TextInput";

const meta = {
  title: "Components/TextInput",
  component: TextInput,
  parameters: {
    layout: "centered",
  },
  tags: ["autodocs"],
  argTypes: {
    value: {
      control: "text",
      description: "The TextInput's value",
    },
    disabled: {
      control: "boolean",
      description: "Is the TextInput disabled?",
    },
    onChange: {
      description: "Called when the input's onChange event triggered",
    },
  },
  args: { onChange: fn() },
} satisfies Meta<typeof TextInput>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Enabled: Story = {
  args: {
    value: 'Enabled',
    disabled: false,
  },
};

export const Disabled: Story = {
  args: {
    value: 'Disabled',
    disabled: true,
  },
};
