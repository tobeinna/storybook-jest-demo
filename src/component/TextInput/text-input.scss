input {
    height: 32px;
    border-top: unset;
    border-right: unset;
    border-left: unset;
    width: -webkit-fill-available;
    outline: none;
}
