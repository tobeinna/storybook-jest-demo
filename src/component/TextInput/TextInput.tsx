import './text-input.scss'

interface TextInputProps {
    value?: string,
    disabled?: boolean,
    onChange: React.Dispatch<React.SetStateAction<string>>
}

function TextInput({ value = '', disabled = false, onChange }: TextInputProps) {
    return <input type="text" value={value} disabled={disabled} onChange={e => onChange(e.target.value)} />;
}

export default TextInput;