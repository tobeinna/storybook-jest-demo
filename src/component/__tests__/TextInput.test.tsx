import { render, screen, fireEvent } from '@testing-library/react';
import TextInput from '../TextInput/TextInput';

describe('TextInput component', () => {
    test('snapshot test for normal text input', () => {
        const tree = render(<TextInput onChange={() => { }} />).container;
        expect(tree).toMatchSnapshot();
    });

    test('snapshot test for disabled text input', () => {
        const tree = render(<TextInput disabled onChange={() => { }} />).container;
        expect(tree).toMatchSnapshot();
    });

    test('renders with correct value', () => {
        render(<TextInput value="Value" onChange={() => { }} />);
        const element = screen.getByDisplayValue(/Value/i);
        expect(element).toBeInTheDocument();
    });

    test('call onChange when value changes', () => {
        const handleChange = jest.fn();
        render(<TextInput value="Initial" onChange={handleChange} />);
        const element = screen.getByDisplayValue(/Initial/i);

        fireEvent.change(element, { target: { value: 'Changed' } });

        expect(handleChange).toHaveBeenCalledWith('Changed');
    });

    test('not call onChange when disabled', () => {
        const handleChange = jest.fn();
        render(<TextInput value="Initial" disabled onChange={handleChange} />);
        const element = screen.getByDisplayValue(/Initial/i);

        fireEvent.change(element, { target: { value: 'Changed' } });

        expect(handleChange).not.toHaveBeenCalled();
    });
});
