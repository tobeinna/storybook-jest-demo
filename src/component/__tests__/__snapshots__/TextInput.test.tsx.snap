// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput component snapshot test for disabled text input 1`] = `
<div>
  <input
    disabled=""
    type="text"
    value=""
  />
</div>
`;

exports[`TextInput component snapshot test for normal text input 1`] = `
<div>
  <input
    type="text"
    value=""
  />
</div>
`;
